nomad/leader.go

```diff
@@ -519,6 +522,52 @@ func (s *Server) periodicUnblockFailedEvals(stopCh chan struct{}) {
 	}
 }
 
+// publishJobSummaryMetrics publishes the job summaries as metrics
+func (s *Server) publishJobSummaryMetrics(stopCh chan struct{}) {
+	// Using a timer instead of a ticker so that we can publish after the
+	// current batch of metrics have been published
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-stopCh:
+			return
+		case <-timer.C:
+			state, err := s.State().Snapshot()
+			if err != nil {
+				timer.Reset(s.config.StatsCollectionInterval)
+				s.logger.Printf("[ERR] nomad: failed to get state: %v", err)
+				continue
+			}
+			ws := memdb.NewWatchSet()
+			iter, err := state.JobSummaries(ws)
+			if err != nil {
+				timer.Reset(s.config.StatsCollectionInterval)
```

Comment:
Is it necessary to reset the timer in each case statement?

Reply:
@chelseakomlo Since we are resetting the timer at the end of the loop, we need to reset it before doing a `continue` in the error branches.

Reply:
If the timing doesn't need to be exact, could you reset the timer at the beginning of the loop for all cases? Otherwise then maybe add comments to explain why timers are set only for some branches.

Reply:
@chelseakomlo I was just thinking since the time taken by the work is not O(1) but O(n) we might cascade a few ticks if the iteration takes longer for some reason like GC pauses or underlying OS issues. I can add comments to explain this.

Reply:
I think it is worth the simpler code to just reset once at the top of the case